dcos/httpclient.go

```diff
@@ -65,18 +54,22 @@ func NewHTTPClient(config *Config) (*http.Client, error) {
 		// Allow http_proxy, https_proxy, and no_proxy.
 		Proxy: http.ProxyFromEnvironment,
 
+		// Set a 10 seconds timeout for the connection to be established.
 		DialContext: (&net.Dialer{
-			Timeout: DefaultHTTPClientDialContextTimeout,
+			Timeout: 10 * time.Second,
 		}).DialContext,
 
-		TLSHandshakeTimeout: DefaultHTTPClientTLSHandshakeTimeout,
+		// Set it to 10 seconds as well for the TLS handshake when using HTTPS.
+		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: config.TLS().Insecure,
 			RootCAs:            config.TLS().RootCAs,
 		},
 
-		MaxIdleConns:        DefaultHTTPClientMaxIdleConns,
-		MaxIdleConnsPerHost: DefaultHTTPClientMaxIdleConnsPerHost,
+		// The client will be dealing with a single host (the one in baseURL),
+		// set max idle connections to 30 regardless of the host.
+		MaxIdleConns:        30,
+		MaxIdleConnsPerHost: 30,
```

Comment:
These constants made the linter fail?

Reply:
It was failing at the place they are defined because the comments weren't starting with eg. `DefaultHTTPClientDialContextTimeout ...`. I think it makes it clearer to have it commented here.

Reply:
But now they're not constants anymore and not exposed in the documentation, right? And you don't see inline comments in the documentation, iirc

Reply:
I've inlined them as I don't see the need to expose these low-level values, strictly speaking they should start with `DefaultTransport`, also given that a DC/OS client interacts with a single host, I find it confusing to have `DefaultHTTPClientMaxIdleConns` and `DefaultHTTPClientMaxIdleConnsPerHost` exposed in the doc.

Reply:
We could lowercase them? I still think it makes sense to define the constant values as constanst 😄

Reply:
I've updated it